product-service/src/main/java/org/arpha/controller/CategoryController.java

```diff
@@ -0,0 +1,47 @@
+package org.arpha.controller;
+
+import com.querydsl.core.types.Predicate;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import lombok.RequiredArgsConstructor;
+import org.arpha.dto.product.CategoryDto;
+import org.arpha.entity.Category;
+import org.arpha.service.CategoryService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.querydsl.binding.QuerydslPredicate;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/categories")
+@SecurityRequirement(name = "Bearer Authentication")
+public class CategoryController {
+
+    private final CategoryService categoryService;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping
+    public CategoryDto create(@RequestBody CategoryDto categoryDto) {
+        return categoryService.createCategory(categoryDto);
+    }
+
+    @GetMapping
+    public Page<CategoryDto> getAllCategories(@QuerydslPredicate(root = Category.class) Predicate predicate, @PageableDefault Pageable pageable) {
+        return categoryService.getAllCategories(predicate, pageable);
+    }
```

Comment:
I suggest adding @PreAuthorize("isAnonymous()") to this endpoint. This should be accessible by not authorized users

Reply:
A bit confused about this part. Why would we need add isAnonymous, if theoretically user and admin could request categories as well, and they would get AccessDeniedException according to documentation.